### DISCUSSION TITLE
Collapse current node/directory

### DISCUSSION THREAD

**Discussion by Akmadan23** [nvim-tree/nvim-tree.lua#2271] [Ideas]
Since when I started using nvim-tree I missed the possibility to collapse a directory having focus on any file inside it instead of focusing the directory itself and then toggling it, which could be tedious for very large directories. The actual function is really simple and straight forward:

```lua
function collapse_current(node)
  if node.open then
    node.open = false
    renderer.draw()
  elseif node.parent then
    node.parent.open = false
    renderer.draw()
    utils.focus_node_or_parent(node)
  end
end
```

My concerns are about what section of the API it belongs to. It could be `api.node.collapse`, `api.tree.collapse_current` or even something else. I'm also not sure about which file should I put the function in, I would probably choose the `lib` module (because it's similar to `lib.expand_or_collapse`), but maybe there is a more appropriate location.
Let me know your considerations so that I can eventually open a PR with all the changes and the updated documentation.

**Comment by gegoune** [comment]
You should be able to achieve that with `require('nvim-tree.api').node.navigate.parent_close()`. Is that what you are after?

**Reply by Akmadan23** [reply]
Oh, I don't know how I managed to completely miss that. I also searched both in the documentation and in the code base but I was looking for the 'collapse' keyword not realizing that the solution was right there... Thank you and sorry for the inconvenience.
